src/gsy_e/models/strategy/scm/external/forecast_mixin.py

```diff
@@ -16,15 +16,31 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-from typing import Dict
+from typing import Dict, TYPE_CHECKING
 
+from gsy_framework.redis_channels import ExternalStrategyChannels
+
+import gsy_e
 from gsy_e.constants import DATE_TIME_FORMAT
 from gsy_e.models.strategy.external_strategies.forecast_mixin import ForecastExternalMixin
 
+if TYPE_CHECKING:
+    from gsy_e.models.area import CoefficientArea
+
 
 class SCMForecastExternalMixin(ForecastExternalMixin):
     """External mixin for forecast strategies in SCM simulations."""
 
+    def activate(self, _area: "CoefficientArea") -> None:
```

Comment:
The CoefficientArea class:
https://github.com/gridsingularity/gsy-e/blob/master/src/gsy_e/models/area/coefficient_area.py#L84
Much more explicit IMO and easier to understand compared to the event-driven mechanism in the normal strategies, however it is confusing because there are different mechanisms for different classes.